### DISCUSSION TITLE
How to use OrderFilter correctly

### DISCUSSION THREAD

**Discussion by amigian74** [api-platform/api-platform#2355] [Q&A] [answered]
Hi,

I'm trying to use the ApiPlatform\Doctrine\Orm\Filter\OrderFilter to get a ordered coolection by name, but it's not working at all. The Filter is applied via annotation like this:

`* @ApiFilter("ApiPlatform\Doctrine\Orm\Filter\OrderFilter", properties={"name"})`

Debugging leads me to the class ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\FilterExtension. But this extension is ignoring the filter because it is not implementing ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\FilterInterface

Am I missing something?

btw: the same happens to the SearchFilter

Thx in advance

**Comment by alanpoulain** [comment, marked as answer]
If you are using API Platform 2.7 in compatibility mode, you are using the wrong filter.
You should use `ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\OrderFilter`.
If you want to use the new filter, you need to disable the compatibility layer: https://api-platform.com/docs/core/upgrade-guide/

**Reply by amigian74** [reply]
Thx for the help- I've tried the old filter too but did'nt work either - now it's working fine.

**Comment by amigian74** [comment]
Sorry to bother again. This works for filtering ans sorting but unfortunately my entities from a different entity manager do'nt work with 'metadata_backward_compatibility_layer' = false

**Reply by amigian74** [reply]
I switched back to version 2.6.8 and everythings works as expected.
